src/sc/sidechain.cpp

```diff
@@ -300,7 +300,7 @@ std::string CSidechain::ToString() const
 }
 
 size_t CSidechain::DynamicMemoryUsage() const {
-    return memusage::DynamicUsage(mImmatureAmounts) + memusage::DynamicUsage(scFees) + memusage::DynamicUsage(scFees_v2);
+    return memusage::DynamicUsage(mImmatureAmounts) + memusage::DynamicUsage(scFees);
```

Comment:
It seems to me `memusage::DynamicUsage(scFees)` would return a `size_t` value computed as `scFees.size()` multiplied by `sizeof(std::shared_ptr<Sidechain::ScFeeData>)`, whilst what we actually want to get is `scFees.size()` multiplied by `sizeof(Sidechain::ScFeeData)`; am I correct?

Reply:
@JackPiri why do you think that ?
From what I see scFees is defined as `std::list<Sidechain::ScFeeData>` (no shared ptr) and the template invoked to compute the allocated size does correctly the multiplication of number of elements in list by size of item type.
Can you expand your doubts ?

Reply:
Ah ok ... so then can we consider this resolved ?

Reply:
This is indeed a very good point. I agree with @JackPiri, the last commit uses `shared_ptr`s, and in its current state the code does not compute dynamic memory correctly.

I was about to fix this by adding proper smart pointers handlers in `src/memusage.h`, but I noticed that in our current codebase most of the things are not computed correctly anyway.

Lines 38-44 state:
```
/** Compute the memory used for dynamically allocated but owned data structures.
 *  For generic data types, this is *not* recursive. DynamicUsage(vector<vector<int> >)
 *  will compute the memory used for the vector<int>'s, but not for the ints inside.
 *  This is for efficiency reasons, as these functions are intended to be fast. If
 *  application data structures require more accurate inner accounting, they should
 *  iterate themselves, or use more efficient caching + updating on modification.
 */
```
I'm not sure the claim on performance reason is valid as of today. Also, I'm not even sure that in general the code in `src/memusage.h` is correct and useful. For example, we make empiric assumptions based on glibc 2.19, but we might run `zend` on very different configurations.

All in all, I see 3 options here:
a) leave everything as it is and stick to the approach suggested in `src/memusage.h`, which means that the mem usage is (slightly) underestimated for this `scFees` list
b) implement accurate measurement for this specific list in `src/sc/sidechain.cpp` for this specific list
c) update `src/memusage.h` to compute mem usage accurately for all stl/template data structures

I personally would go for option `a` in this PR, but definitely go for `c` in another PR.

What do you guys think?